### ISSUE TITLE
Terminal Payment: Overcharge for non-paid refunded orders 

### ISSUE THREAD

**Issue by htdat** [Automattic/woocommerce-payments#2846]
### Describe the bug

I can describe this issue mentioned here https://github.com/Automattic/woocommerce-payments/issues/2390#issuecomment-877199723 when watching @malinajirka does on his Woo Mobile app and virtual reader/terminal. 

### To Reproduce

1. Customer: In the front-end of the website, try to checkout with a COD order:  a tshirt ($10) and a hoodie($15) => total $25.
2. Merchant:  in WP Admin, see the status of this order: Processing.
3. Customer: Come to the store and want to remove tshirt from the order ($10). 
4. Merchant: Edit this order in WP Admin, refund tshirt ($10). 
5. Merchant: Open Woo Mobile, grab customer’s credit card, and put it to the terminal, and try to charge. 

During step 5, Woo Mobile sends a POST request to `/payments/orders/<id>/capture_terminal_payment`, which is implemented here https://github.com/Automattic/woocommerce-payments/pull/1689

### Actual behavior

WooMobile should charge $25.

### Expected behavior

Expect: WooMobile should charge $15 only. 

### Possible solutions? 

- We may check the code and charge $15 only = Total - Refunded Amount = Remaining Amount. 
- OR we do not allow this flow, and whenever refunding exists in an order, we do not process `capture_chage`. 

Relevant code: 

https://github.com/Automattic/woocommerce-payments/blob/103fabc96ceceeadf4e9127cd8ed006c5965b50f/includes/class-wc-payment-gateway-wcpay.php#L1583-L1585

**Comment by htdat** [comment]
@kidinov - can help confirm my steps above?

**Comment by kidinov** [comment]
> Merchant: Edit this order in WP Admin, refund tshirt ($10).

Hey @htdat 👋.  It sounds about right, but in my case, I was refunding via mobile app, not WP admin. Probably doesn't make any difference

**Comment by htdat** [comment]
Update: @malinajirka will raise an internal discussion regarding this - ref p1630921922049800/1630591782.032900-slack-C01B82V67DL

**Comment by malinajirka** [comment]
One more clarification note to this issue:

```
Actual behavior
WooMobile should charge $25.
```
This is imho not true - that's the point of this issue. Even when the mobile app charges "$15", WCPay thinks the app tried to charge $25 - Stripe's dashboard shows $15 as expected but WCPay keeps showing $25.

Here are screenshots with a similar example where the app tried to charge $35 dollars but WCPay thought it's charging $80.
Stripe's dashboard: (Notice it says $35)
<img width="1444" alt="Screenshot 2021-09-08 at 12 03 30" src="https://user-images.githubusercontent.com/2261188/132490129-0cbf9aa0-dea7-41b1-af4e-47fd491a75fd.png">

WCPay/WPadmin: (Notice it says $80)
<img width="1384" alt="Screenshot 2021-09-08 at 12 03 45" src="https://user-images.githubusercontent.com/2261188/132490205-b3b796be-d059-475d-9213-aac23ba87f3e.png">

**Comment by htdat** [comment]
> > Actual behavior
WooMobile charges $25.
> 
> This is imho not true - that's the point of this issue. Even when the mobile app charges "$15", WCPay thinks the app tried to charge $25 - Stripe's dashboard shows $15 as expected but WCPay keeps showing $25.

I am describing how it works from what I understand now. That said, IMHO, now it's more important to decide if we allow this flow. 
From that, we can tweak the current WCPay (client) code if necessary. See my heading `Possible solutions` above.

**Comment by htdat** [comment]
Similar to https://github.com/Automattic/woocommerce-payments/issues/2390#issuecomment-915148758, the approach for this issue is based on pdfdoF-8k-p2

**Comment by a-danae** [comment]
@htdat, looks like we're going with the second solution, is that correct? Referring to this one:
> OR we do not allow this flow, and whenever refunding exists in an order, we do not process capture_chage.

I've read through the referenced resources but I'd like to confirm which path we're taking before getting started on this.

**Comment by htdat** [comment]
> I've read through the referenced resources but I'd like to confirm which path we're taking before getting started on this.

Yes, I've left a comment in our internal discussion pdfdoF-8k-p2#comment-367 and updated the issue here.

**Comment by htdat** [comment]
@a-danae - in case, you're not aware, I think one PR will fix both this issue and #2390.

**Comment by a-danae** [comment]
Nice, got it. Thanks for the heads-up! 🙂

**Comment by kalessil** [comment]
@dmvrtx please make a decision on how to proceed here.
